### DISCUSSION TITLE
Language Server Protocol in JupyterLite

### DISCUSSION THREAD

**Discussion by bollwyvl** [jupyterlite/jupyterlite#895] [Ideas]
During the December 2022 JupyterLite workshop, I'm looking into a `jupyterlite-lsp` shim.

The first pass at this might:
- be a `serverlite` extension
- use `mock-socket` to fool the labextension into thinking there's a normal websocket server
- offer a `addLanguageServerSpec(id, {spec})`
- demonstrate this with a couple language servers (e.g. `json`, `yaml`)

I am still of the mind that full kernels might be better off being able to host language servers in-loop over comm, but that's a story for another day.

**Comment by bollwyvl** [comment]
Repo started (very broken): https://github.com/jupyterlite/jupyterlite-lsp

**Comment by qqdaiyu55** [comment]
@bollwyvl Just curious of the future planning of `jupyterlite-lsp` thx? Automatic completion & code intelligence are extremely important for coding experience.
